docs/docs/specs/v0/composer-2.md

```diff
@@ -0,0 +1,389 @@
+- **Vault**: HALO universal PWA which stores identity and data
+- **Stack**: A user experience paradigm (component) which enables editing columns of mixed content with arbitrary (extensible) content types. A key feature is the draggability of stack elements to re-order them.
+
+## Non-definitions
+
+- **Frames**: are not defined because explicit coupling to ECHO is a non-goal, and without that Frames are just regular React Components which can assume certain contexts or props within this application framework. React Components or Components are used in place of Frames from prior art.
+
+## Out of scope
+
+- Distribution (or registration) of Components separately from Plugins. Plugins are the only unit of registration and distribution and contain / provide a list of Components to the host application. To identify a component globally, a string containing the identity of the plugin _and_ the component must be used.
+
+## Applications are made of Surfaces and Components
+
+- The application is a component `<App />`.
+- A component can have any number of `<Surface />` as children inside it.
+- A Surface figures out what Component(s) to render given a set of `loaded plugins`, a `data context`, and `other control props`. It can be understood as the the developer intention to render/visualize the given `data context` (datum) using whatever appropriate component as provided by Plugins and configured by the the end user and/or the developer.
+- The main `<App />` component defines a single, full-screen `<Surface />` for rendering "anything". i.e.: does not opine about what it is, delegates that to plugins and initial configuration entirely.
+- The main `<App />` component also wraps the main `Surface` with an `AppContext` provider representing the total and entire state of the application which becomes available through a hook to any nested component.
+
+## Example Structure
+
+Below is a diagram illustrating how:
+
+- a full application is built up to handle a `Stack` of `Composer` and `Image`
+- `Surfaces` are the only element of abstraction necessary to both compose the entire application chrome and implement the abstract portion of `Stack`.
+
+![Composer Structural Diagram](./composer.drawio.svg)
+
+## Stacks
+
+- Stacks are just components which rely on the same `<Surface />` concept to delegate the concrete rendering of stack elements to components provided by plugins.
+
+## Routing
+
+- Routing (the main router and top level routes) are declared outside (around) the `<App />` component and allow the `App` to pass route information as state in the total `AppState` context for all descendants.
+- Nested components inside `<App />` are equally free to use `<Route />` elements if necessary as well.
+- Nested components are encouraged to perform navigation by submitting `Actions` as opposed to calling `navigate()` directly, although situations like `<Button to="/some/route"/>` should also be supported.
```

Comment:
I don't think the interface for the app’s route should be distinct from any other part of the `AppState` – maybe that is already what this section intends to communicate, but I'd phrase some things this way:

> components are welcome to use subcomponents like `<Route … />` inasmuch as `Route` is like any other subcomponent that selects for specific `AppState` conditions, only focused on the `AppState`’s route substate.

> components can “navigate” by dispatching actions that affect `AppState`’s route substate.